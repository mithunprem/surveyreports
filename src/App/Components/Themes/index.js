import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Row } from 'reactstrap';
import ThemeDetails from '../ThemeDetails';

const Themes = ({ themes }) => {
  return (
    <Fragment>
      <hr />
      <h6 className="ml-4 mb-2"> Themes Focussed :</h6>
      <Row className="themes ml-1 mb-3 mr-1">
        {
          themes.map((theme, index) => {
            return <ThemeDetails key={index} theme={theme}/>
          })
        }
      </Row>
    </Fragment>
  );
}

Themes.propTypes = {
  themes: PropTypes.array
}

export default Themes;
