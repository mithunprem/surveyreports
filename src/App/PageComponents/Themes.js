import React, { Fragment } from 'react';
import { Row } from 'reactstrap';
import ThemeDetails from './ThemeDetails';

const Themes = ({ themes }) => {
  return (
    <Fragment>
      <hr />
      <h6 className="ml-4 mb-2"> Themes Focussed :</h6>
      <Row className="theme-details ml-1 mb-3 mr-1">
        {
          themes.map((theme, index) => {
            return <ThemeDetails key={index} theme={theme}/>
          })
        }
      </Row>
    </Fragment>
  );
}

export default Themes;
