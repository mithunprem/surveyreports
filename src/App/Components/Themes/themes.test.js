import React from 'react';
import { create } from 'react-test-renderer';
import Themes from '../../Components/Themes';

describe("Themes", () => {

  const themes = [
    {
      name: 'The Work',
      questions: [
        { description: 'Test question 1',
          survey_responses: [
            {id: 1, question_id: 4, respondent_id: 1, response_content: "4"},
            {id: 2, question_id: 4, respondent_id: 2, response_content: "4"}
          ],
          averageRating: 4 }
      ],
      averageRating: 4.5
    },
    {
      name: 'The Place',
      questions: [
        { description: 'Test question 2',
          survey_responses: [
            {id: 1, question_id: 5, respondent_id: 1, response_content: "4"},
            {id: 2, question_id: 5, respondent_id: 2, response_content: "4"}
          ],
          averageRating: 4 }
      ],
      averageRating: 3.5
    }
  ];


  test("should render without throwing an error", () => {
    const themeDetails = create(<Themes themes={themes} />).toJSON();
    expect(themeDetails).toMatchSnapshot();
  });
});
