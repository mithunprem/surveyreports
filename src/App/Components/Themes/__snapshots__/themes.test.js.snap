// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themes should render without throwing an error 1`] = `
Array [
  <hr />,
  <h6
    className="ml-4 mb-2"
  >
     Themes Focussed :
  </h6>,
  <div
    className="themes ml-1 mb-3 mr-1 row"
  >
    <div
      className="mb-4 col-12 col-sm-12 col-md-6 col-xl-6"
    >
      <div
        className="theme-details margin-green card card-body"
      >
        <h5
          className="card-header"
        >
          The Work
        </h5>
        <div
          className="card-body"
        >
          <button
            aria-label={null}
            className="btn btn-link btn-sm"
            onClick={[Function]}
            type="button"
          >
            View responses in detail
          </button>
          <div
            className="collapse"
            style={Object {}}
          >
            <table
              className="questions-table table table-striped"
            >
              <thead>
                <tr>
                  <th>
                    Question
                  </th>
                  <th
                    className="average-rating-header"
                  >
                    Average Rating
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Test question 1
                  </td>
                  <td
                    className="average-rating-value text-green"
                  >
                    4.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="text-muted card-footer"
        >
          Average rating : 
          4.50
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-sm-12 col-md-6 col-xl-6"
    >
      <div
        className="theme-details margin-green card card-body"
      >
        <h5
          className="card-header"
        >
          The Place
        </h5>
        <div
          className="card-body"
        >
          <button
            aria-label={null}
            className="btn btn-link btn-sm"
            onClick={[Function]}
            type="button"
          >
            View responses in detail
          </button>
          <div
            className="collapse"
            style={Object {}}
          >
            <table
              className="questions-table table table-striped"
            >
              <thead>
                <tr>
                  <th>
                    Question
                  </th>
                  <th
                    className="average-rating-header"
                  >
                    Average Rating
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Test question 2
                  </td>
                  <td
                    className="average-rating-value text-green"
                  >
                    4.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          className="text-muted card-footer"
        >
          Average rating : 
          3.50
        </div>
      </div>
    </div>
  </div>,
]
`;
